src/main/java/com/conveyal/r5/profile/Path.java

```diff
@@ -24,13 +27,43 @@
     // It does effectively allow you to leave out the boardStopPositions and alightStopPositions, but a subclass could do that too.
     public int[] patterns;
     public int[] boardStops;
+    // Note that including stop arrival times would imply a distinct path for each departure minute.
     public int[] alightStops;
     public int[] alightTimes;
     public int[] trips;
     public int[] boardStopPositions;
     public int[] alightStopPositions;
+    public StreetMode accessMode;
+    /**
+     * Used only in propagation for writing paths for Taui outputs
+     * TODO update egress mode outside of path (e.g. in wrapper or array parallel to perIterationTravelTimes in
+     *  propagater)
+     */
+    public StreetMode egressMode;
+
+    // Additional characteristics (not keys for hashing)
+    public int inVehicleTime;
+    public int waitTime;
```

Comment:
The new class IterationDetails has these same fields. If I understand right every IterationDetails associated with a Path will have the same inVehicleTime and access/egress time as that path, just different initial wait times (and therefore different total times).

What if instead of a set of IterationDetails, each Path instead had a map from iterations (or departure times) to initial waits? To me this seems to more concisely factor out only the information that changes from one iteration's path to another. It could also eliminate the nonstandard inclusion of fields that are not hashed in Path, which may cause problems for code comprehension in the future.

It would also be possible to include this iteration-to-wait-time map directly as a field in Path, leaving it null when not needed. That's the composition approach I was thinking of, but maybe the Multimap is better if for a reason I'm not thinking of.